RELEASE_NOTES.md

```diff
@@ -20,6 +20,56 @@ Notable changes include:
   * Bug fixes/improvements:
 
 
+Version 2025.03.0 -- Release date 2025-03-dd
```

Comment:
Don't forget to update the date.

Reply:
I assume dd here should be 0 as its the patch version?

Reply:
I set the date for Monday 3/17 to be conservative. Need to get CI passing -- I bumped alloc time to avoid timeouts on lassen.